lib/facter/gluster.rb

```diff
@@ -5,18 +5,27 @@
 volume_ports = {}
 
 binary = Facter.value('gluster_custom_binary')
-if not binary or not File.executable? binary
+if !binary || !File.executable?(binary)
   # http://stackoverflow.com/questions/2108727/which-in-ruby-checking-if-program-exists-in-path-from-ruby/5471032#5471032
   exts = ENV['PATHEXT'] ? ENV['PATHEXT'].split(';') : ['']
   ENV['PATH'].split(File::PATH_SEPARATOR).each do |path|
-    exts.each { |ext|
+    exts.each do |ext|
       exe = File.join(path, "gluster#{ext}")
       binary = exe if File.executable? exe
-    }
+    end
   end
 end
 
-if binary then
+def volume_list
+  return @volume_list if @volume_list
```

Comment:
IMHO it would be cleaner to only have one return statement in a method. what about:
unless @volume_list {
....
}
return @volume_list